main.go

```diff
@@ -35,6 +35,28 @@ func readFromFile(filePath string) (string, error) {
 	return string(content), nil
 }
 
+// parseContent determines the content string based on the flags and arguments
+func parseContent(directText *string, args []string) (string, error) {
```

Comment:
Handling each case in parseContent separately for clarity is indeed a valid point now. Since, I've already merged the changes from @karimalzalek I think a new PR will be required here, maybe titled: refactor: update parseContent function for improved readability

Reply:
I'm glad you liked my suggestion